---
title: "Sieradzki_soil_wetup_succession"
author: "Ella Sieradzki"
date: "2023-04-05"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE)

library("DESeq2")
library("Biobase")
library("ggplot2")
library("data.table")
library("heatmap3")
library("tidyverse")
library("tibble")
library("vegan")
library("gplots")
library("compositions")
library("ape")
library("cowplot")
library("plotly")
library("purrr")
library("reshape2")
library("MASS")
library("ggpubr")
library("stringr")
library(rstatix)
library(broom)
library("RColorBrewer")
```



```{r}
# Load data
# Load MAG taxonomy file
tax <- read.csv("MAG_tax.csv", header=T, stringsAsFactors = F)
tax$Bin <- gsub(".fa", "", tax$Bin)

# DON'T LOAD THIS UNLESS YOU RUN DESEQ Load Rsubread output for DESeq
infile = "Rsubread_output.csv"
mtx_counts = read.csv(infile, header = TRUE, row.names = 1, stringsAsFactors = F)
colnames(mtx_counts) <- gsub("_unf_init.bam", "", colnames(mtx_counts))
colnames(mtx_counts) <- gsub("12C16O", "", colnames(mtx_counts))
mtx_counts$bin <- gsub("_k[0-9].*", "", rownames(mtx_counts))
# Keep only significantly enriched MAGs (CI.L>0)
enr_MAGs <- c()
enr_MAGs_100 <- read.csv("AFE_MAGs_CIL_tax_100.csv", stringsAsFactors = F, header=T)
enr_MAGs <- unique(enr_MAGs_100$bin)
enr_MAGs_50 <- read.csv("AFE_MAGs_CIL_tax_50.csv", stringsAsFactors = F, header=T)
enr_MAGs <- c(enr_MAGs, unique(enr_MAGs_50$bin))
enr_MAGs_tax <- tax[tax$Bin %in% unique(enr_MAGs),]
enr_MAGs_tax$phylum <- gsub(".*p__", "", gsub(";c.*", "", enr_MAGs_tax$classification))
enr_MAGs_tax$class <- gsub(".*c__", "", gsub(";o.*", "", enr_MAGs_tax$classification))
write.csv(enr_MAGs_tax, "enr_MAGs_all_tax.csv")[,-1]
enr_MAGs_tax$Bin <- gsub("\\.", "_", gsub("-", "_", enr_MAGs_tax$Bin))

# DON'T LOAD THIS UNLESS YOU RUN DESEQ Analyze only genes from isotopically enriched MAGs
mtx_counts_enr <- mtx_counts[mtx_counts$bin %in% enr_MAGs_tax$Bin,]
mtx_counts <- mtx_counts_enr[,-ncol(mtx_counts_enr)]
rownames(mtx_matrix) = rownames(mtx_counts)
mtx_matrix[is.na(mtx_matrix)] = 0
mtx_matrix = mtx_matrix[,order(colnames(mtx_matrix))]

# DRAM annotations
# CAZy distilled functions (use flag --distillate_gene_names when running DRAM distill)
ann_cazy <- read.csv("distill_MAGs_enr_all_metabolism_CAZy.csv",header=T,stringsAsFactors = F)

# kofam annotations
ann <- read.delim("annotations.tsv", sep="\t", header=T, stringsAsFactors = F, strip.white=TRUE)
ann <- ann[ann$kegg_id != "" | ann$peptidase_id != "" | ann$cazy_hits != "",]
# DRAM adds the bin name again - remove it
ann$X <- gsub("^12.*_12", "12", ann$X)
ann$X <- gsub("^H.*_H", "H", ann$X)

designfile = "mtx_design_ecofun_genomes4.txt" 

mtx_design = read.delim(designfile, header = TRUE)
mtx_design <- mtx_design[!(rownames(mtx_design) %in% c("H3_Rhizo_39", "H1_RhizoLitter_2", "H2_RhizoLitter_9")),] # Remove these because these columns are mostly 0s, was messing up the geometric mean 

# To reload the saved RDS from DESeq2:
res_all_data_bulkcomp_sig <- readRDS("res_enrMAGs_sig.rds")
res_all_data_bulkcomp_sig$X <- rownames(res_all_data_bulkcomp_sig)

# To load the saved dataframes of genes with significant differential abundance
res_all_data_bulkcomp_sig_only <- read.csv("DESeq2_sigonly_MAGs_enr.csv",stringsAsFactors = F, strip.white=TRUE)
res_all_data_bulkcomp_sig_only_ann <- read.csv("res_all_data_bulkcomp_sig_only_MAGs_enr.csv",stringsAsFactors = F, strip.white=TRUE)[,-1]
```



```{r}
# Pie chart of MAG taxonomy
ggcols <- read.csv("~/Dropbox/PostDoc/4th_wedge/MAGs_disrat0.33/class_ggcolors.csv", stringsAsFactors = F)
enr_MAGs_tax <- enr_MAGs_tax[order(enr_MAGs_tax$phylum),]
p_freq <- as.data.frame(matrix(ncol=2, nrow=length(unique(enr_MAGs_tax$phylum))))
colnames(p_freq) <- c("Phylum","count")
i=1
for (p in unique(enr_MAGs_tax$phylum)) {
  p_freq$Phylum[i] <- p
  p_freq$count[i] <- nrow(tax[enr_MAGs_tax$phylum==p,])
  i=i+1
}
c_freq <- as.data.frame(matrix(ncol=2, nrow=length(unique(enr_MAGs_tax$class))))
colnames(c_freq) <- c("Class","count")
i=1
for (c in unique(enr_MAGs_tax$class)) {
  c_freq$Class[i] <- c
  c_freq$count[i] <- nrow(enr_MAGs_tax[enr_MAGs_tax$class==c,])
  i=i+1
}
c_freq <- c_freq %>% mutate(rand=1)
c_freq <- merge(c_freq, unique(enr_MAGs_tax[,3:4]), by.x="Class",by.y="class")
c_freq <- merge(c_freq, ggcols, by="Class", all.x=T)
c_freq <- c_freq[order(c_freq$phylum),]
library(plotrix)
pie(p_freq$count, labels=p_freq$Phylum, col=rainbow(nrow(p_freq)), main="Enriched MAG collection by phylum")
ggplot(c_freq, aes(fill=Class, y=count, x=rand)) + 
  geom_col(position="stack", color="black")

```

```{r}
# Running DESeq2: comparing by condition (combination of time+prec)
dds2 = DESeqDataSetFromMatrix(countData = mtx_matrix, colData = mtx_design, design = ~ condition)
dds2 = dds2[ rowSums(counts(dds2)) > 5, ] # Filter out rows with < 5 counts
design(dds2) = ~ 0+condition
dds2 = DESeq(dds2)
res2 = results(dds2)
resultsNames(dds2)
saveRDS(dds2, "dds_traits_4thwedge_enr.rds")
dds2_counts <- counts(dds2, normalized=TRUE)
write.csv(dds2_counts, "dds2_counts_enr.csv", row.names = F)
rownames(dds2_counts) <- rownames(res2)
dds2_ann <- merge(dds2_counts, ann, by.x="row.names", by.y="X")
write.csv(dds2_ann, "dds2_counts_ann_enr.csv", row.names = F)

# Compare between consecutive time points
res = res2
resultsNames(dds2)
res2 = results(dds2, contrast = c("condition", "24h_100", "0h_100")) # group, numerator, denominator
comparison = "24h_100_vs_0h_100"
res_sig = which(res2$padj < 0.05)
length(res_sig)
baseMean_df = data.frame(res2$baseMean)
betas_df = data.frame(res2$log2FoldChange)
stat_df = data.frame(res2$stat)
padj_df = data.frame(res2$padj)
setnames(baseMean_df, "res2.baseMean", paste0("baseMean_", comparison))
setnames(betas_df, "res2.log2FoldChange", paste0("betas_", comparison))
setnames(stat_df, "res2.stat", paste0("stat_", comparison))
setnames(padj_df, "res2.padj", paste0("padj_", comparison))

res2 = results(dds2, contrast = c("condition", "48h_100", "24h_100")) # group, numerator, denominator
comparison = "48h_100_vs_24h_100"
res_sig = which(res2$padj < 0.05)
length(res_sig)
baseMean_df = data.frame(res2$baseMean)
betas_df = data.frame(betas_df, res2$log2FoldChange)
stat_df = data.frame(stat_df, res2$stat)
padj_df = data.frame(padj_df, res2$padj)
setnames(baseMean_df, "res2.baseMean", paste0("baseMean_", comparison))
setnames(betas_df, "res2.log2FoldChange", paste0("betas_", comparison))
setnames(stat_df, "res2.stat", paste0("stat_", comparison))
setnames(padj_df, "res2.padj", paste0("padj_", comparison))

res2 = results(dds2, contrast = c("condition", "72h_100", "48h_100")) # group, numerator, denominator
comparison = "72h_100_vs_48h_100"
res_sig = which(res2$padj < 0.05)
length(res_sig)
baseMean_df = data.frame(res2$baseMean)
betas_df = data.frame(betas_df, res2$log2FoldChange)
stat_df = data.frame(stat_df, res2$stat)
padj_df = data.frame(padj_df, res2$padj)
setnames(baseMean_df, "res2.baseMean", paste0("baseMean_", comparison))
setnames(betas_df, "res2.log2FoldChange", paste0("betas_", comparison))
setnames(stat_df, "res2.stat", paste0("stat_", comparison))
setnames(padj_df, "res2.padj", paste0("padj_", comparison))

res2 = results(dds2, contrast = c("condition", "168h_100", "72h_100"))
comparison = "168h_100_vs_72h_100"
res_sig = which(res2$padj < 0.05)
length(res_sig)
baseMean_df = data.frame(res2$baseMean)
betas_df = data.frame(betas_df, res2$log2FoldChange)
stat_df = data.frame(stat_df, res2$stat)
padj_df = data.frame(padj_df, res2$padj)
setnames(baseMean_df, "res2.baseMean", paste0("baseMean_", comparison))
setnames(betas_df, "res2.log2FoldChange", paste0("betas_", comparison))
setnames(stat_df, "res2.stat", paste0("stat_", comparison))
setnames(padj_df, "res2.padj", paste0("padj_", comparison))

res2 = results(dds2, contrast = c("condition", "48h_50", "0h_50"))
comparison = "48h_50_vs_0h_50"
res_sig = which(res2$padj < 0.05)
length(res_sig)
baseMean_df = data.frame(res2$baseMean)
betas_df = data.frame(betas_df, res2$log2FoldChange)
stat_df = data.frame(stat_df, res2$stat)
padj_df = data.frame(padj_df, res2$padj)
setnames(baseMean_df, "res2.baseMean", paste0("baseMean_", comparison))
setnames(betas_df, "res2.log2FoldChange", paste0("betas_", comparison))
setnames(stat_df, "res2.stat",paste0("stat_", comparison))
setnames(padj_df, "res2.padj", paste0("padj_", comparison))

res2 = results(dds2, contrast = c("condition", "168h_50", "48h_50"))
comparison = "168h_50_vs_48h_50"
res_sig = which(res2$padj < 0.05)
length(res_sig)
baseMean_df = data.frame(res2$baseMean)
betas_df = data.frame(betas_df, res2$log2FoldChange)
stat_df = data.frame(stat_df, res2$stat)
padj_df = data.frame(padj_df, res2$padj)
setnames(baseMean_df, "res2.baseMean", paste0("baseMean_", comparison))
setnames(betas_df, "res2.log2FoldChange", paste0("betas_", comparison))
setnames(stat_df, "res2.stat",paste0("stat_", comparison))
setnames(padj_df, "res2.padj", paste0("padj_", comparison))

res2 = results(dds2, contrast = c("condition", "0h_50", "0h_100"))
comparison = "0h_50_vs_0h_100"
res_sig = which(res2$padj < 0.05)
length(res_sig)
baseMean_df = data.frame(res2$baseMean)
betas_df = data.frame(betas_df, (res2$log2FoldChange))
stat_df = data.frame(stat_df, res2$stat)
padj_df = data.frame(padj_df, res2$padj)
setnames(baseMean_df, "res2.baseMean", "baseMean_0h_50_vs_0h_100")
setnames(betas_df, colnames(betas_df)[ncol(betas_df)], "betas_0h_50_vs_0h_100")
setnames(stat_df, "res2.stat","stat_0h_50_vs_0h_100")
setnames(padj_df, "res2.padj", "padj_0h_50_vs_0h_100")

res2 = results(dds2, contrast = c("condition", "48h_100", "48h_50"))
comparison = "48h_100_vs_48h_50"
res_sig = which(res2$padj < 0.05)
length(res_sig)
baseMean_df = data.frame(res2$baseMean)
betas_df = data.frame(betas_df, res2$log2FoldChange)
stat_df = data.frame(stat_df, res2$stat)
padj_df = data.frame(padj_df, res2$padj)
setnames(baseMean_df, "res2.baseMean", paste0("baseMean_", comparison))
setnames(betas_df, "res2.log2FoldChange", paste0("betas_", comparison))
setnames(stat_df, "res2.stat",paste0("stat_", comparison))
setnames(padj_df, "res2.padj", paste0("padj_", comparison))

res2 = results(dds2, contrast = c("condition", "168h_100", "168h_50"))
comparison = "168h_100_vs_168h_50"
res_sig = which(res2$padj < 0.05)
length(res_sig)
baseMean_df = data.frame(res2$baseMean)
betas_df = data.frame(betas_df, res2$log2FoldChange)
stat_df = data.frame(stat_df, res2$stat)
padj_df = data.frame(padj_df, res2$padj)
setnames(baseMean_df, "res2.baseMean", paste0("baseMean_", comparison))
setnames(betas_df, "res2.log2FoldChange", paste0("betas_", comparison))
setnames(stat_df, "res2.stat",paste0("stat_", comparison))
setnames(padj_df, "res2.padj", paste0("padj_", comparison))

res_all_data_bulkcomp = data.frame(betas_df, stat_df, padj_df)
rownames(res_all_data_bulkcomp) = rownames(res)
colnames(res_all_data_bulkcomp)[7] <- "betas_0h_50_vs_0h_100"
res_all_data_bulkcomp$Bin <- gsub("_k[0-9].*", "", rownames(res_all_data_bulkcomp))
saveRDS(res_all_data_bulkcomp, file = paste(working_directory,"/res_enrMAGs.rds", sep = ""))

# Only keep p values where the comparison by Wald test was significantly different
padj_matrix = as.matrix(res_all_data_bulkcomp[,19:27]) # This is essentially padj_df, but only saved res_all_data, so working from that
padj_matrix[is.na(padj_matrix)] = 1 # Change NA to 1 so will get filtered out in next step
padj_matrix[which(padj_matrix > 0.05)] = NA # Convert all p values > 0.05 to NA
res_all_data_bulkcomp_sig = data.frame(res_all_data_bulkcomp[,c(1:9,28)], as.data.frame(padj_matrix)) # Remake res_all_data with only significant p values, all non-significant p values are NA
res_all_data_bulkcomp_sig$X <- rownames(res_all_data_bulkcomp_sig)

saveRDS(res_all_data_bulkcomp_sig, file = paste(working_directory,"/res_enrMAGs_sig.rds", sep = ""))

tax$Bin <- gsub("\\.", "_", gsub("-","_", tax$Bin))
res_all_data_bulkcomp_sig <- merge(res_all_data_bulkcomp_sig, tax, by = "Bin")
write.csv(res_all_data_bulkcomp_sig, file = paste0(working_directory,"results_enrMAGs_sig.csv"), row.names = F)

# subset to rows that have at least one significant p value (<0.05)
res_all_data_bulkcomp_sig_only <- subset(res_all_data_bulkcomp_sig, !(is.na(padj_24h_100_vs_0h_100) & (is.na(padj_48h_100_vs_24h_100)) & (is.na(padj_72h_100_vs_48h_100)) & (is.na(padj_168h_100_vs_72h_100)) & (is.na(padj_168h_100_vs_72h_100)) & (is.na(padj_48h_50_vs_0h_50)) & (is.na(padj_168h_50_vs_48h_50)) & (is.na(padj_0h_50_vs_0h_100)) & (is.na(padj_48h_100_vs_48h_50)) & (is.na(padj_168h_100_vs_168h_50))))
nrow(res_all_data_bulkcomp_sig_only)
write.csv(res_all_data_bulkcomp_sig_only, "DESeq2_sigonly_MAGs_enr.csv", row.names = F)
res_all_data_bulkcomp_sig_only_ann <- merge(res_all_data_bulkcomp_sig_only, ann, by="X", all.x=T)
write.csv(res_all_data_bulkcomp_sig_only_ann, "res_all_data_bulkcomp_sig_only_MAGs_enr.csv", row.names = F)
```

```{r}
# Beta diversity matrix
dds2_counts <- read.csv("dds2_counts_enr.csv") %>% column_to_rownames("X")
dds2_ann <- read.csv("dds2_counts_ann_enr.csv")

dds2_bray_dm = vegdist(t(dds2_counts), method="bray")
dds2_dist = as.dist(dds2_bray_dm)

# PCoA
pcoa_bray <- pcoa(dds2_dist)
jac_variances <- data.frame(pcoa_bray$values$Relative_eig) %>% 
  rownames_to_column(var = "PCaxis") %>% 
  data.frame
pcoa_bray_df <- data.frame(pcoa_bray$vectors) %>% 
  rownames_to_column(var = "sample") %>% 
  data.frame

pcoa_bray_df$sample <- gsub("_unf_init.bam", "", gsub("X_", "", pcoa_bray_df$sample))
pcoa_bray_df$grp <- gsub("_P[0-9].*$", "", pcoa_bray_df$sample)
pcoa_bray_df$plotnum <- gsub(".*P","P", pcoa_bray_df$sample)
pcoa_bray_df$time <- gsub("h.*", "", pcoa_bray_df$grp)
pcoa_bray_df$time <- factor(pcoa_bray_df$time, levels=c(0,24,48,72,168))
eigenvalues<-round(jac_variances[,2], digits = 4)*100

ggplot(pcoa_bray_df) +
  geom_point(aes(x = Axis.1, y = Axis.2, color=time, shape=plotnum), size = 4) +
  ylab(paste0('Co 2 ',eigenvalues[2],'%')) + #Extract y axis value from variance
  xlab(paste0('Co 1 ',eigenvalues[1],'%')) + #Extract x axis value from variance
  ggtitle('Log-Ratio PCoA Ordination') +
  coord_fixed(ratio = 1) +
  theme_bw() + scale_color_manual(values=rev(brewer.pal(name="Paired", n=6))) +
  scale_shape_manual(values = c(15, 0, 1, 2, 16, 17))
ggsave("fig2_panelB_PCoA.pdf")
```

```{r}
# Get significantly enriched MAGs (lower confidence interval limit > 0) either in 100% or 50% treatment
enr_MAGs <- c()
enr_MAGs_100 <- read.csv("AFE_MAGs_CIL_tax_100.csv", stringsAsFactors = F, header=T)
enr_MAGs <- unique(enr_MAGs_100$bin)
enr_MAGs_50 <- read.csv("AFE_MAGs_CIL_tax_50.csv", stringsAsFactors = F, header=T)
enr_MAGs <- c(enr_MAGs, unique(enr_MAGs_50$bin))
afe100 <- enr_MAGs_100[,c(which(colnames(enr_MAGs_100) %like% "AFE"), which(colnames(enr_MAGs_100) == "bin"))]
afe100 <- unique(afe100)
afe50 <- enr_MAGs_50[,c(which(colnames(enr_MAGs_50) %like% "AFE"), which(colnames(enr_MAGs_50) == "bin"))]
afe50 <- unique(afe50)
afe <- merge(afe100, afe50, by="bin", all=T)
afe[is.na(afe)] <- 0
colnames(afe) <- c("bin", "AFE_24h", "AFE_48h", "AFE_72h", "AFE_168h", "AFE_48h_50", "AFE_168h_50")
```

```{r}
# Create supplemental table S1 - genome info, RPKM and median AFE
sts1 <- read.csv("coverm_rpkm_wFirmicutes_corrected.csv", header=T, stringsAsFactors = F) %>%
  dplyr::rename(bin = Genome) %>%
  left_join(afe[, c(2, which(colnames(afe) %like% "AFE"))]) %>%
  dplyr::rename(Bin = bin) %>%
  left_join(tax, by="Bin") %>%
  rename_at(vars(starts_with("X12C16O")), funs(sub("X12C16O", "RPKM", .))) %>%
  dplyr::rename(MAG = Bin) %>%
  left_join(mag_info)
afe_cols <- which(colnames(sts1) %like% "AFE")
rpkm_cols <- which(colnames(sts1) %like% "RPKM")
sts1$Detected <- FALSE
sts1$Detected[rowSums(sts1[,rpkm_cols]) > 0] <- TRUE
sts1$Growing <- FALSE
sts1$Growing[sts1$AFE_168h>0 | sts1$AFE_168h_50>0 | sts1$AFE_48h>0 | sts1$AFE_48h_50>0 | sts1$AFE_24h>0 | sts1$AFE_72h>0] <- TRUE
write.table(sts1, "../Traits_paper/sup_tbl_S1_RPKM_AFE_info.tsv", sep="\t", row.names = F)
```

```{r}
# How many growing organisms are enriched in multiple time points?
temp2 <- rpkm_enr %>%
  dplyr::rename(bin = Genome) %>%
  left_join(afe, by="bin")
temp2$AFE_168h[temp2$AFE_168h > 0] <- 1 
temp2$AFE_168h_50[temp2$AFE_168h_50 > 0] <- 1
temp2$AFE_48h[temp2$AFE_48h > 0] <- 1
temp2$AFE_48h_50[temp2$AFE_48h_50 > 0] <- 1
temp2$AFE_24h[temp2$AFE_24h > 0] <- 1
temp2$AFE_72h[temp2$AFE_72h > 0] <- 1
temp2$AFE_168h[temp2$AFE_168h < 0] <- 0
temp2$AFE_168h_50[temp2$AFE_168h_50 < 0] <- 0
temp2$AFE_48h[temp2$AFE_48h < 0] <- 0
temp2$AFE_48h_50[temp2$AFE_48h_50 < 0] <- 0
temp2$AFE_24h[temp2$AFE_24h < 0] <- 0
temp2$AFE_72h[temp2$AFE_72h < 0] <- 0

count(rowSums(temp2[,c("AFE_24h", "AFE_48h", "AFE_72h", "AFE_168h")]) > 1) # 63
count(rowSums(temp2[,c("AFE_48h_50", "AFE_168h_50")]) > 1) # 30
```

```{r}
# Figure 2 panel A: RPKM regression
rpkm <- read.delim("coverm_RPKM.txt", header=T, stringsAsFactors = F, sep="\t")
rpkm <- rpkm[rowSums(rpkm[,2:25])>0,]
rpkm_melt <- rpkm %>% pivot_longer(!Genome, names_to="sample", values_to="rpkm")
rpkm_melt$sample <- gsub("_unf_init.RPKM", "", rpkm_melt$sample)
rpkm_melt$enr <- "no"
rpkm_melt$enr[rpkm_melt$Genome %in% afe$bin] <- "yes"
rpkm_melt$time <- as.numeric(gsub("X12C16O_", "", gsub("h_.*", "", rpkm_melt$sample)))
rpkm_melt <- rpkm_melt %>% mutate(treat = case_when(
  rpkm_melt$sample %like% "_50" ~ "RP",
  TRUE ~ "MP"
))
rpkm_mean <- rpkm_melt %>%
  group_by(time, treat, enr) %>%
  summarise(rpkm_mean = mean(rpkm), rpkm_sd = sd(rpkm)) %>%
  mutate(rpkm_se = rpkm_sd/sqrt(3))
ggplot(data=rpkm_mean, aes(x=time, y=rpkm_mean, colour=enr)) + theme_classic() + 
  geom_line() +
  geom_point()+
  geom_errorbar(aes(ymin=rpkm_mean-rpkm_se, ymax=rpkm_mean+rpkm_se), width=.2,
                 position=position_dodge(0.05)) +
  xlab("Time (h)") + ylab("Reads per kilo million (RPKM)") 
ggsave("fig2_panelA_RPKM.pdf")
```


```{r}
# Figure 4
# Make vector of colors for heatmap
nHalf = 50
rc1 <- colorRampPalette(colors = c("blue", "white"), space="Lab")(nHalf)    
rc2 <- colorRampPalette(colors = c("white","lightgoldenrod1","red"), space="Lab")(nHalf)
rampcols <- c(rc1, rc2)

afe <- merge(afe, tax, by.x="bin", by.y="Bin")
afe$Phylum <- gsub(".*;p__", "", gsub(";c.*", "", afe$classification))
afe$Class <- gsub(".*;c__", "", gsub(";o.*", "", afe$classification))
afe <- merge(afe, ggcols, by="Class")
afe_cols <- which(colnames(afe) %like% "AFE")
afe <- afe[order(afe$Phylum, afe$Color,decreasing = T),]
# Draw AFE heatmap
heatmap3(as.matrix(afe[,afe_cols]), col=rampcols, cexCol = 1, cexRow=0.2, labRow = afe$bin, Rowv = NA, Colv=NA, RowSideColors = afe$Color)
# Create RPKM heatmap for figure 3
rpkm <- read.delim("coverm_RPKM.txt", header=T, stringsAsFactors = F, sep="\t")
rpkm <- rpkm[rowSums(rpkm[,2:25])>0,]
rpkm_melt <- rpkm %>% pivot_longer(!Genome, names_to="sample", values_to="rpkm")
rpkm_melt$sample <- gsub("_unf_init.RPKM", "", rpkm_melt$sample)

rpkm_enr <- rpkm[rpkm$Genome %in% afe$bin,]
rpkm_enr <- rpkm_enr[match(afe$bin, rpkm_enr$Genome),]
ord <- c("X12C16O_0h_100", "X12C16O_24h_100", "X12C16O_48h_100", "X12C16O_72h_100", "X12C16O_168h_100", "X12C16O_0h_50", "X12C16O_48h_50", "X12C16O_168h_50")
heatmap3(as.matrix(rpkm_enr[,ord]), col=rc2, cexCol = 1, cexRow=0.2, labRow = rpkm_enr$Genome, Rowv = NA, Colv=NA, margins=c(0,0))
leg <- unique(afe[,c("Class", "Color")])
plot.new()
legend("topleft", legend=leg$Class, fill=leg$Color, xpd=T, inset=c(0,-0.2))
```


```{r}
# Figure 3
# Create Venn diagrams of significant genes from time points / treatments
library("ggvenn")
x <- list(
  as.character(res_all_data_bulkcomp_sig_only_ann[!is.na(res_all_data_bulkcomp_sig_only_ann$padj_24h_100_vs_0h_100),1]),
  as.character(res_all_data_bulkcomp_sig_only_ann[!is.na(res_all_data_bulkcomp_sig_only_ann$padj_48h_100_vs_24h_100),1]),
  as.character(res_all_data_bulkcomp_sig_only_ann[!is.na(res_all_data_bulkcomp_sig_only_ann$padj_72h_100_vs_48h_100),1]),
  as.character(res_all_data_bulkcomp_sig_only_ann[!is.na(res_all_data_bulkcomp_sig_only_ann$padj_168h_100_vs_72h_100),1])
)
names(x) <- c("24h", "48h", "72h", "168h")
ggvenn(x)
x <- list(
  res_all_data_bulkcomp_sig_only_ann[!is.na(res_all_data_bulkcomp_sig_only_ann$padj_48h_50_vs_0h_50),1],
  res_all_data_bulkcomp_sig_only_ann[!is.na(res_all_data_bulkcomp_sig_only_ann$padj_168h_50_vs_48h_50),1]
)
names(x) <- c("48h", "168h")
ggvenn(x)
x <- list(
  res_all_data_bulkcomp_sig_only_ann[!is.na(res_all_data_bulkcomp_sig_only_ann$padj_0h_50_vs_0h_100),1],
  res_all_data_bulkcomp_sig_only_ann[!is.na(res_all_data_bulkcomp_sig_only_ann$padj_48h_100_vs_48h_50),1],
  res_all_data_bulkcomp_sig_only_ann[!is.na(res_all_data_bulkcomp_sig_only_ann$padj_168h_100_vs_1),1]
)
names(x) <- c("0h", "48h", "168h")
ggvenn(x)

# subset to rows that have at least one significant p value (<0.05) in the 100% treatment
res_all_data_bulkcomp_sig_only_100 <- subset(res_all_data_bulkcomp_sig_only, !(is.na(padj_24h_100_vs_0h_100) & (is.na(padj_48h_100_vs_24h_100)) & (is.na(padj_72h_100_vs_48h_100)) & (is.na(padj_168h_100_vs_72h_100)) & (is.na(padj_168h_100_vs_72h_100))))
nrow(res_all_data_bulkcomp_sig_only_100)
effect_size_100 <- c(
nrow(subset(res_all_data_bulkcomp_sig_only_100, !(is.na(padj_24h_100_vs_0h_100)))),
nrow(subset(res_all_data_bulkcomp_sig_only_100, !(is.na(padj_48h_100_vs_24h_100)))),
nrow(subset(res_all_data_bulkcomp_sig_only_100, !(is.na(padj_72h_100_vs_48h_100)))),
nrow(subset(res_all_data_bulkcomp_sig_only_100, !(is.na(padj_168h_100_vs_72h_100)))))
names(effect_size_100) <- c("24_vs_0", "48_vs_24", "72_vs_48", "168_vs_72")

# subset to rows that have at least one significant p value (<0.05) in the 50% treatment
res_all_data_bulkcomp_sig_only_50 <- subset(res_all_data_bulkcomp_sig_only, !((is.na(padj_48h_50_vs_0h_50)) & (is.na(padj_168h_50_vs_48h_50))))
nrow(res_all_data_bulkcomp_sig_only_50)
effect_size_50 <- c(
  nrow(subset(res_all_data_bulkcomp_sig_only_100, !(is.na(padj_48h_50_vs_0h_50)))),
  nrow(subset(res_all_data_bulkcomp_sig_only_100, !(is.na(padj_168h_50_vs_48h_50)))))
names(effect_size_50) <- c("48_vs_0", "168_vs_48")

# subset to rows that have at least one significant p value (<0.05) when comparing treatments
res_all_data_bulkcomp_sig_only_treat <- subset(res_all_data_bulkcomp_sig, !((is.na(padj_0h_50_vs_0h_100)) & (is.na(padj_48h_100_vs_48h_50)) & (is.na(padj_168h_100_vs_168h_50))))
nrow(res_all_data_bulkcomp_sig_only_treat)
effect_size_treat <- c(
nrow(subset(res_all_data_bulkcomp_sig_only_treat, !((is.na(padj_0h_50_vs_0h_100))))),
nrow(subset(res_all_data_bulkcomp_sig_only_treat, !((is.na(padj_48h_100_vs_48h_50))))),
nrow(subset(res_all_data_bulkcomp_sig_only_treat, !((is.na(padj_168h_100_vs_168h_50))))))
names(effect_size_treat) <- c("0", "48", "168")

# Figure 3
#effect_size_all <- c(effect_size_100, effect_size_50, effect_size_treat)
pdf(file = "fig3_legacy_genes.pdf")
barplot(log10(1+effect_size_treat), ylab="Log10 number of significant differentially abundant genes", las=2, col="purple")
dev.off()
```

```{r}
# Supplementary figure 2
# Volcano plots of differential abundance
p24 <- ggplot(data=betas, aes(x=betas_24h_100_vs_0h_100, y=-log10(padj_24h_100_vs_0h_100))) + 
  geom_point() + theme_classic() + ggtitle("24h") + 
  ylab("-log10 adjujsted p-value") + xlab("log2 fold change") + 
  geom_vline(xintercept=c(-1, 1), col="red")
p48 <- ggplot(data=betas, aes(x=betas_48h_100_vs_24h_100, y=-log10(padj_48h_100_vs_24h_100))) + 
  geom_point() + theme_classic() + ggtitle("48h") + 
  ylab("-log10 adjujsted p-value") + xlab("log2 fold change") + 
  geom_vline(xintercept=c(-1, 1), col="red")
p72 <- ggplot(data=betas, aes(x=betas_72h_100_vs_48h_100, y=-log10(padj_72h_100_vs_48h_100))) + 
  geom_point() + theme_classic() + ggtitle("72h") + 
  ylab("-log10 adjujsted p-value") + xlab("log2 fold change") + 
  geom_vline(xintercept=c(-1, 1), col="red")
p168 <- ggplot(data=betas, aes(x=betas_168h_100_vs_72h_100, y=-log10(padj_168h_100_vs_72h_100))) + 
  geom_point() + theme_classic() + ggtitle("168h") + 
  ylab("-log10 adjujsted p-value") + xlab("log2 fold change") + 
  geom_vline(xintercept=c(-1, 1), col="red")
ggarrange(p24, p48, p72, p168, nrow=2, ncol=2)

p48 <- ggplot(data=betas, aes(x=betas_48h_50_vs_0h_50, y=-log10(padj_48h_50_vs_0h_50))) + 
  geom_point() + theme_classic() + ggtitle("48h 50%") + 
  ylab("-log10 adjujsted p-value") + xlab("log2 fold change") + 
  geom_vline(xintercept=c(-1, 1), col="red")
p168 <- ggplot(data=betas, aes(x=betas_168h_50_vs_48h_50, y=-log10(padj_168h_50_vs_48h_50))) + 
  geom_point() + theme_classic() + ggtitle("168h 50%") + 
  ylab("-log10 adjujsted p-value") + xlab("log2 fold change") + 
  geom_vline(xintercept=c(-1, 1), col="red")
ggarrange(p48, p168, nrow=1, ncol=2)

p0 <- ggplot(data=betas, aes(x=betas_0h_50_vs_0h_100, y=-log10(padj_0h_50_vs_0h_100))) + 
  geom_point() + theme_classic() + ggtitle("0h legacy") + 
  ylab("-log10 adjujsted p-value") + xlab("log2 fold change") + 
  geom_vline(xintercept=c(-1, 1), col="red")
p48 <- ggplot(data=betas, aes(x=betas_48h_100_vs_48h_50, y=-log10(padj_48h_100_vs_48h_50))) + 
  geom_point() + theme_classic() + ggtitle("48h legacy") + 
  ylab("-log10 adjujsted p-value") + xlab("log2 fold change") + 
  geom_vline(xintercept=c(-1, 1), col="red")
p168 <- ggplot(data=betas, aes(x=betas_168h_100_vs_168h_50, y=-log10(padj_168h_100_vs_168h_50))) + 
  geom_point() + theme_classic() + ggtitle("168h legacy") + 
  ylab("-log10 adjujsted p-value") + xlab("log2 fold change") + 
  geom_vline(xintercept=c(-1, 1), col="red")
ggarrange(p0, p48, p168, nrow=1, ncol=3)

# Violin plots of significant betas - sup. fig. 2
plot.new()
par(mfrow=c(2, 4))
min_betas <- min(betas$betas_0h_50_vs_0h_100)
max_betas <- max(betas$betas_48h_100_vs_48h_50)
violin_plot(betas$betas_24h_100_vs_0h_100[!is.na(betas$padj_24h_100_vs_0h_100)], main="24h 100%", ylim=c(min_betas, max_betas))
violin_plot(betas$betas_48h_100_vs_24h_100[!is.na(betas$padj_48h_100_vs_24h_100)], main="48h 100%", ylim=c(min_betas, max_betas))
violin_plot(betas$betas_72h_100_vs_48h_100[!is.na(betas$padj_72h_100_vs_48h_100)], main="72h 100%", ylim=c(min_betas, max_betas))
violin_plot(betas$betas_168h_100_vs_72h_100[!is.na(betas$padj_168h_100_vs_72h_100)], main="168h 100%", ylim=c(min_betas, max_betas))
violin_plot(betas$betas_48h_50_vs_0h_50[!is.na(betas$padj_48h_50_vs_0h_50)], main="48h 50%", ylim=c(min_betas, max_betas))
violin_plot(betas$betas_168h_50_vs_48h_50[!is.na(betas$padj_168h_50_vs_48h_50)], main="168h 50%", ylim=c(min_betas, max_betas))
violin_plot(betas$betas_0h_50_vs_0h_100[!is.na(betas$padj_0h_50_vs_0h_100)], main="0h legacy", ylim=c(min_betas, max_betas))
violin_plot(betas$betas_48h_100_vs_48h_50[!is.na(betas$padj_48h_100_vs_48h_50)], main="48h legacy", ylim=c(min_betas, max_betas))
```

```{r}
# Functions in enriched MAGs - 79 with AFE confidence interval lower limit>
enr_ann <- read.delim("distill_MAGs_enr_all.tsv",sep="\t", header=T, stringsAsFactors = F)
enr_ann_afe <- merge(enr_ann, afe, by.x="genome", by.y="bin")
colnames(enr_ann_afe)
enr_ann_afe[enr_ann_afe == "True"] <- 1
enr_ann_afe[enr_ann_afe == "False"] <- 0
cazy_cols <- which(colnames(enr_ann_afe) %like% "CAZ")
n_cols <- which(colnames(enr_ann_afe) %like% "Nitrogen")
s_cols <- which(colnames(enr_ann_afe) %like% "Sulfur")
```


```{r}
# Figure 6
# CAZy genes and pathways
ann_cazy_long <- melt(ann_cazy, id.vars=1:5)
ann_cazy_long <- ann_cazy_long %>% filter(value!="")
# DRAM adds the bin name to the beginning of the fasta header. Remove that.
ann_cazy_long$value <- sub("^.*_12", "12", ann_cazy_long$value)
ann_cazy_long$value <- sub("^.*_H", "H", ann_cazy_long$value)
colnames(ann_cazy_long)[6] <- "Bin"
colnames(ann_cazy_long)[7] <- "X"
ann_cazy_long$Bin <- gsub("-", "_", gsub("\\.", "_",ann_cazy_long$Bin))
betas_CAZy <- merge(betas, ann_cazy_long)
colnames(betas_CAZy)[2] <- "ORF"
betas_grouped_CAZy <- betas_CAZy %>% group_by(subheader) %>% summarize(across(starts_with("betas"), ~ mean(.x, na.rm = TRUE)))

# CAZy abundance changes between time points
cazy_df <- c()
plot.new()
par(mfrow=c(3,2))
temp <- colSums(enr_ann_afe[enr_ann_afe$AFE_24h_100_col != "lightgrey",cazy_cols])
cazy_df$h24 <- temp

ord <- order(-temp)
temp <- colSums(enr_ann_afe[enr_ann_afe$AFE_72h_100_col != "lightgrey",cazy_cols])
cazy_df$h72 <- temp

temp <- colSums(enr_ann_afe[enr_ann_afe$AFE_48h_100_col != "lightgrey",cazy_cols])
cazy_df$h48 <- temp

temp <- colSums(enr_ann_afe[enr_ann_afe$AFE_168h_100_col != "lightgrey",cazy_cols])
cazy_df$h168 <- temp

temp <- colSums(enr_ann_afe[enr_ann_afe$AFE_48h_50_col != "lightgrey",cazy_cols])
cazy_df$h48_50 <- temp

temp <- colSums(enr_ann_afe[enr_ann_afe$AFE_168h_50_col != "lightgrey",cazy_cols])
cazy_df$h168_50 <- temp

cazy_df <- data.frame(cazy_df)
cazy_df$gene <- gsub("CAZy  ", "", gsub("\\.", " ", rownames(cazy_df)))
cazy_df_melt <- melt(cazy_df)
kruskal.test(value~variable, data=cazy_df_melt) # there is at least one statistically significant group
k24_48 <- kruskal.test(value~variable, data=cazy_df_melt[cazy_df_melt$variable %in% c("h24", "h48"),]) #p=0.02
k48_72 <- kruskal.test(value~variable, data=cazy_df_melt[cazy_df_melt$variable %in% c("h48", "h72"),]) #p=8.9e-05
k72_168 <- kruskal.test(value~variable, data=cazy_df_melt[cazy_df_melt$variable %in% c("h72", "h168"),]) #p=0.04
k48_168 <- kruskal.test(value~variable, data=cazy_df_melt[cazy_df_melt$variable %in% c("h48_50", "h168_50"),]) #p=0.32
k48 <- kruskal.test(value~variable, data=cazy_df_melt[cazy_df_melt$variable %in% c("h48", "h48_50"),]) #p=8.9e-05
k168 <- kruskal.test(value~variable, data=cazy_df_melt[cazy_df_melt$variable %in% c("h168", "h168_50"),]) #p=0.25
# Extracting p-values of all comparisons
p_vals <- c(k24_48$p.value, k48_72$p.value, k72_168$p.value, k48_168$p.value, k48$p.value, k168$p.value)
# Correcting for multiple comparisons with Benjamini-Hochberg
p_vals_BH <- p.adjust(p_vals, "BH", length(p_vals)) # 0.002, 0.00009, 0.04, 0.32, 0.00009, 0.25

temp1 <- enr_ann_afe %>%
  filter_at(vars(starts_with("AFE")), any_vars(. > 0)) %>%
  select(matches('CAZy|col$')) %>%
  pivot_longer(!ends_with("col"), names_to = "pathway", values_to = "count") %>%
  mutate(pathway = gsub("CAZy\\.\\.", "", pathway)) %>%
  mutate(pathway = gsub("\\.*$", "", pathway)) %>%
  mutate(pathway_short = case_when(
    pathway == "Alpha.galactans" ~ "Ag",
    pathway == "Alpha.mannan" ~ "Am",
    pathway == "Amorphous.Cellulose" ~ "Ac",
    pathway == "Arabinan" ~ "Ar",
    pathway == "Arabinose.cleavage" ~ "Ae",
    pathway == "Beta.galactan..pectic.galactan" ~ "Bg",
    pathway == "Beta.mannan" ~ "Bm",
    pathway == "Chitin" ~ "Ch",
    pathway == "Crystalline.Cellulose" ~ "Cc",
    pathway == "Fucose.Cleavage" ~ "Fu",
    pathway == "Mixed.Linkage.glucans" ~ "Ml",
    pathway == "Mucin" ~ "Mu",
    pathway == "Pectin" ~ "Pe",
    pathway == "Polyphenolics" ~ "Po",
    pathway == "Rhamnose.cleavage" ~ "Rh",
    pathway == "Starch" ~ "St",
    pathway == "Sulf.Polysachharides" ~ "Sp",
    pathway == "Xylans" ~ "Xs",
    pathway == "Xyloglucan" ~ "Xn"
  )) %>%
  mutate(pathway_short = factor(pathway_short, levels=c("Ch", "Am", "Ae", "Rh", "Fu", "Mu", "Po", "Ar", "Pe", "Xs", 
                                                        "Xn", "St", "Bm", "Cc", "Ac", "Bg", "Ag", "Ml", "Sp"))) %>%
  mutate(grp = factor(case_when(
    pathway_short %in% c("Am", "Ch") ~ "Fungi",
    pathway_short %in% c("Ae", "Rh", "Fu") ~ "Mono",
    pathway_short == "Mu" ~ "Mucin",
    pathway_short %in% c("Ag", "Ml", "Sp", "Bg") ~ "Poly",
    TRUE ~ "Plant"
  ), levels = c("Fungi", "Mono", "Mucin", "Plant", "Poly"))) %>%
  mutate(col = case_when(
    pathway_short == "Am" ~ "#FEE391",
    pathway_short == "Ch" ~ "#FEC44F",
    pathway_short == "Ae" ~ "#2171B5",
    pathway_short == "Rh" ~ "#6BAED6",
    pathway_short == "Fu" ~ "#BDD7E7",
    pathway_short == "Mu" ~ "#FF00FF",
    pathway_short == "Po" ~ "#00441B",
    pathway_short == "Ar" ~ "#006D2C",
    pathway_short == "Pe" ~ "#238B45",
    pathway_short == "Xs" ~ "#41AB5D",
    pathway_short == "Xn" ~ "#74C476",
    pathway_short == "St" ~ "#A1D99B",
    pathway_short == "Bm" ~ "#C7E9C0",
    pathway_short == "Cc" ~ "#E5F5E0",
    pathway_short == "Ac" ~ "#F7FCF5",
    pathway_short == "Bg" ~ "#A50F15",
    pathway_short == "Ag" ~ "#DE2D26",
    pathway_short == "Ml" ~ "#FB6A4A",
    pathway_short == "Sp" ~ "#FCAE91"
  ))

temp24 <- temp1 %>%
  filter(AFE_24h_100_col != "lightgrey") %>%
  group_by(pathway_short, grp, col) %>%
  summarise_at("count", sum) %>%
  mutate(count = count*100/nrow(enr_ann_afe[enr_ann_afe$AFE_24h_100_col != "lightgrey",])) %>%
  arrange(grp, pathway_short)

temp48 <- temp1 %>%
  filter(AFE_48h_100_col != "lightgrey") %>%
  group_by(pathway_short, grp, col) %>%
  summarise_at("count", sum) %>%
  mutate(count = count*100/nrow(enr_ann_afe[enr_ann_afe$AFE_48h_100_col != "lightgrey",])) %>%
  arrange(grp, pathway_short) %>%
  left_join(temp24[,c("pathway_short", "count")], by="pathway_short") %>%
  mutate(dif = count.x-count.y)
p48 <- ggplot(temp48, aes(x=pathway_short, y=dif, fill=col)) + 
  geom_bar(stat = "identity", color="black", width=0.8, position = position_dodge(width=0.2)) + theme_classic() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) +
  scale_fill_identity() +
  ylim(c(-12,12)) + xlab(NULL) + ylab(NULL) + ggtitle("48h vs. 24h")

temp48 <- temp1 %>%
  filter(AFE_48h_100_col != "lightgrey") %>%
  group_by(pathway_short, grp, col) %>%
  summarise_at("count", sum) %>%
  mutate(count = count*100/nrow(enr_ann_afe[enr_ann_afe$AFE_48h_100_col != "lightgrey",])) %>%
  arrange(grp, pathway_short)
temp72 <- temp1 %>%
  filter(AFE_72h_100_col != "lightgrey") %>%
  group_by(pathway_short, grp, col) %>%
  summarise_at("count", sum) %>%
  mutate(count = count*100/nrow(enr_ann_afe[enr_ann_afe$AFE_72h_100_col != "lightgrey",])) %>%
  arrange(grp, pathway_short) %>%
  left_join(temp48[,c("pathway_short", "count")], by="pathway_short") %>%
  mutate(dif = count.x-count.y)
p72 <- ggplot(temp72, aes(x=pathway_short, y=dif, fill=col)) + 
  geom_bar(stat = "identity", color="black", width=0.8, position = position_dodge(width=0.2)) + theme_classic() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) +
  scale_fill_identity() +
  ylim(c(-12,12)) + xlab(NULL) + ylab(NULL) + ggtitle("72h vs. 48h")

temp72 <- temp1 %>%
  filter(AFE_72h_100_col != "lightgrey") %>%
  group_by(pathway_short, grp, col) %>%
  summarise_at("count", sum) %>%
  mutate(count = count*100/nrow(enr_ann_afe[enr_ann_afe$AFE_72h_100_col != "lightgrey",])) %>%
  arrange(grp, pathway_short)
temp168 <- temp1 %>%
  filter(AFE_168h_100_col != "lightgrey") %>%
  group_by(pathway_short, grp, col) %>%
  summarise_at("count", sum) %>%
  mutate(count = count*100/nrow(enr_ann_afe[enr_ann_afe$AFE_168h_100_col != "lightgrey",])) %>%
  arrange(grp, pathway_short) %>%
  left_join(temp72[,c("pathway_short", "count")], by="pathway_short") %>%
  mutate(dif = count.x-count.y)
p168 <- ggplot(temp168, aes(x=pathway_short, y=dif, fill=col)) + 
  geom_bar(stat = "identity", color="black", width=0.8, position = position_dodge(width=0.2)) + theme_classic() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) +
  scale_fill_identity() +
  ylim(c(-12,12)) + xlab(NULL) + ylab(NULL) + ggtitle("168h vs. 72h")

temp48 <- temp1 %>%
  filter(AFE_48h_100_col != "lightgrey") %>%
  group_by(pathway_short, grp, col) %>%
  summarise_at("count", sum) %>%
  mutate(count = count*100/nrow(enr_ann_afe[enr_ann_afe$AFE_48h_100_col != "lightgrey",])) %>%
  arrange(grp, pathway_short)
temp48_50 <- temp1 %>%
  filter(AFE_48h_50_col != "lightgrey") %>%
  group_by(pathway_short, grp, col) %>%
  summarise_at("count", sum) %>%
  mutate(count = count*100/nrow(enr_ann_afe[enr_ann_afe$AFE_48h_50_col != "lightgrey",])) %>%
  arrange(grp, pathway_short) %>%
  left_join(temp48[,c("pathway_short", "count")], by="pathway_short") %>%
  mutate(dif = count.x-count.y)
p48_50_treat <- ggplot(temp48_50, aes(x=pathway_short, y=dif, fill=col)) + 
  geom_bar(stat = "identity", color="black", width=0.8, position = position_dodge(width=0.2)) + theme_classic() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) +
  scale_fill_identity() +
  ylim(c(-12,12)) + xlab(NULL) + ylab(NULL) + ggtitle("48h 50% vs. 48h 100%")

temp48_50 <- temp1 %>%
  filter(AFE_48h_50_col != "lightgrey") %>%
  group_by(pathway_short, grp, col) %>%
  summarise_at("count", sum) %>%
  mutate(count = count*100/nrow(enr_ann_afe[enr_ann_afe$AFE_48h_50_col != "lightgrey",])) %>%
  arrange(grp, pathway_short)
temp168_50 <- temp1 %>%
  filter(AFE_168h_50_col != "lightgrey") %>%
  group_by(pathway_short, grp, col) %>%
  summarise_at("count", sum) %>%
  mutate(count = count*100/nrow(enr_ann_afe[enr_ann_afe$AFE_168h_50_col != "lightgrey",])) %>%
  arrange(grp, pathway_short) %>%
  left_join(temp48_50[,c("pathway_short", "count")], by="pathway_short") %>%
  mutate(dif = count.x-count.y)
p168_50_time <- ggplot(temp168_50, aes(x=pathway_short, y=dif, fill=col)) + 
  geom_bar(stat = "identity", color="black", width=0.8, position = position_dodge(width=0.2)) + theme_classic() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) +
  scale_fill_identity() +
  ylim(c(-12,12)) + xlab(NULL) + ylab(NULL) + ggtitle("168h 50% vs. 48 50%")

temp168 <- temp1 %>%
  filter(AFE_168h_100_col != "lightgrey") %>%
  group_by(pathway_short, grp, col) %>%
  summarise_at("count", sum) %>%
  mutate(count = count*100/nrow(enr_ann_afe[enr_ann_afe$AFE_168h_100_col != "lightgrey",])) %>%
  arrange(grp, pathway_short)
temp168_50 <- temp1 %>%
  filter(AFE_168h_50_col != "lightgrey") %>%
  group_by(pathway_short, grp, col) %>%
  summarise_at("count", sum) %>%
  mutate(count = count*100/nrow(enr_ann_afe[enr_ann_afe$AFE_168h_50_col != "lightgrey",])) %>%
  arrange(grp, pathway_short) %>%
  left_join(temp168[,c("pathway_short", "count")], by="pathway_short") %>%
  mutate(dif = count.x-count.y)
p168_50_treat <- ggplot(temp168_50, aes(x=pathway_short, y=dif, fill=col)) + 
  geom_bar(stat = "identity", color="black", width=0.8, position = position_dodge(width=0.2)) + theme_classic() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) +
  scale_fill_identity() +
  ylim(c(-12,12)) + xlab(NULL) + ylab(NULL) + ggtitle("168h 50% vs. 168h 100%")

ggarrange(p48, p48_50_treat, p72, p168_50_treat, p168, p168, p168_50_time, ncol=2, nrow=4)
ggsave(filename="fig6_CAZy_manual_ord_delta.pdf", width=7.5, height=6)
```

```{r}
# Supplementary figure 3
# CAZy per MAG

# Compare completeness between growers and detected
mag_info <- read.csv("genomeInformation_derep.csv", stringsAsFactors = F)
mag_info$MAG <- gsub("\\.fa", "", mag_info$MAG)
enr_mags <- read.csv("enr_MAGs_all_tax.csv")[,-1] %>% select("Bin")
unenr_mags <- unenr_ann$genome
enr_comp <- mag_info$completeness[mag_info$MAG %in% enr_mags$Bin]
unenr_comp <- mag_info$completeness[mag_info$MAG %in% unenr_mags]
t.test(enr_comp, unenr_comp)

# Compare completeness between fast and slow growers
mag_info_us <- mag_info %>%
  mutate(MAG = gsub("-", "_", gsub("\\.", "_", MAG)))
afe_us <- afe %>%
  mutate(bin = gsub("-", "_", gsub("\\.", "_", bin)))
fast_grow <- scan("../Traits_paper/fast_growers.txt", sep="\n", what = "charater")
fast_grow <- gsub("-", "_", gsub("\\.", "_", fast_grow))
slow_grow <- afe_us$bin[!(afe_us$bin %in% fast_grow)]
slow_grow <- gsub("-", "_", gsub("\\.", "_", slow_grow))
fast_comp <- mag_info_us$completeness[mag_info_us$MAG %in% fast_grow]
slow_comp <- mag_info_us$completeness[mag_info_us$MAG %in% slow_grow]
t.test(fast_comp, slow_comp)

all_cazy <- read.csv("all_MAGs_metabolism_summary_cazy.csv") %>%
  filter(header=="CAZY") %>%
  select(!c("gene_description", "module", "header", "subheader", "fasta")) %>%
  column_to_rownames("gene_id") %>%
  t() %>% data.frame() %>% rownames_to_column("genome")
all_cazy$genome <- gsub("^X", "", all_cazy$genome)
all_cazy$genome <- gsub("_H.*", "", all_cazy$genome)
fast_grow <- gsub("-", "\\.", fast_grow)
slow_grow <- gsub("-", "\\.", slow_grow)

all_cazy_us <- all_cazy %>%
  mutate(genome = gsub("\\.", "_", gsub("-", "_", genome)))
fast_grow_cazy <- all_cazy_us %>%
  filter(genome %in% fast_grow) %>%
  column_to_rownames("genome") 
fast_grow_cazy[fast_grow_cazy != ""] <- 1
fast_grow_cazy[fast_grow_cazy == ""] <- 0 
fast_grow_cazy <- fast_grow_cazy %>% mutate_all(as.numeric)

slow_grow_cazy <- all_cazy_us %>%
  filter(genome %in% slow_grow) %>%
  column_to_rownames("genome") 
slow_grow_cazy[slow_grow_cazy != ""] <- 1
slow_grow_cazy[slow_grow_cazy == ""] <- 0 
slow_grow_cazy <- slow_grow_cazy %>% mutate_all(as.numeric)

fgc_sum <- rowSums(fast_grow_cazy)
fsc_sum <- rowSums(slow_grow_cazy)
t.test(fgc_sum, fsc_sum, alternative="less") #p=0.02, t=-2
mean(fgc_sum) #33
mean(fsc_sum) #41

# Plot number of cazy genes per genome by AFWE per time point
tax_us <- tax %>%
  mutate(Bin = gsub("-", "_", gsub("\\.", "_", Bin)))
fc <- data.frame(c(fgc_sum, fsc_sum), row.names=c(names(fgc_sum), names(fsc_sum))) %>%
  rownames_to_column("bin") %>%
  left_join(afe_us) %>%
  dplyr::rename(Bin = bin) %>%
  left_join(tax_us) %>%
  dplyr::rename(cazy_num = c.fgc_sum..fsc_sum.) %>%
  mutate(phylum = factor(gsub(".*p__", "", gsub(";c__.*", "", classification))))
tax_cols  <- brewer.pal(length(unique(fc$phylum)), "Paired")
names(tax_cols) <- unique(fc$phylum)
temp <- fc[fc$AFE_24h>0,]
p24 <- ggplot(data=temp, aes(x=AFE_24h, y=cazy_num, color=phylum)) + theme_classic() + geom_point() + geom_vline(xintercept = 0.2) + scale_color_manual(values = tax_cols)
temp <- fc[fc$AFE_48h>0,]
p48 <- ggplot(data=temp, aes(x=AFE_48h, y=cazy_num, color=phylum)) + theme_classic() + geom_point() + geom_vline(xintercept = 0.2) + scale_color_manual(values = tax_cols)
temp <- fc[fc$AFE_72h>0,]
p72 <- ggplot(data=temp, aes(x=AFE_48h, y=cazy_num, color=phylum)) + theme_classic() + geom_point() + geom_vline(xintercept = 0.2) + scale_color_manual(values = tax_cols)
temp <- fc[fc$AFE_168h>0,]
p168 <- ggplot(data=temp, aes(x=AFE_168h, y=cazy_num, color=phylum)) + theme_classic() + geom_point() + geom_vline(xintercept = 0.2) + scale_color_manual(values = tax_cols)
temp <- fc[fc$AFE_48h_50>0,]
p48_50 <- ggplot(data=temp, aes(x=AFE_48h_50, y=cazy_num, color=phylum)) + theme_classic() + geom_point() + geom_vline(xintercept = 0.2) + scale_color_manual(values = tax_cols)
temp <- fc[fc$AFE_168h_50>0,]
p168_50 <- ggplot(data=temp, aes(x=AFE_168h_50, y=cazy_num, color=phylum)) + theme_classic() + geom_point() + geom_vline(xintercept = 0.2) + scale_color_manual(values = tax_cols)
ggarrange(p24,p72,p48,p168,p48_50,p168_50, nrow=3, ncol=2)
ggsave("../Traits_paper/sup_fig_3.pdf")
```


```{r}
# Supplementary figure 4
# CAZy counts and differential abundance in enriched, unenriched but detected and undetected MAGs
# The DRAM output has names of CAZy genes that I can pull from betas rows
enr_ann <- read.delim("distill_MAGs_enr_all.tsv", sep="\t", header=T, stringsAsFactors = F)
unenr_ann <- read.delim("detected_MAGs_product.tsv", sep="\t", header=T, stringsAsFactors = F)
all_ann <- read.delim("all_MAGs_product.tsv", sep="\t", header=T, stringsAsFactors = F)
cazy_cols <- which(colnames(enr_ann) %like% "CAZ")
enr_ann[enr_ann == "True"] <- 1
enr_ann[enr_ann == "False"] <- 0
unenr_ann[unenr_ann == "True"] <- 1
unenr_ann[unenr_ann == "False"] <- 0
all_ann[all_ann == "True"] <- 1
all_ann[all_ann == "False"] <- 0
enr_ann[,cazy_cols] <- apply(enr_ann[,cazy_cols], 1, as.numeric)
unenr_ann[,cazy_cols] <- apply(unenr_ann[,cazy_cols], 1, as.numeric)
all_ann[,cazy_cols] <- apply(all_ann[,cazy_cols], 1, as.numeric)

cazy_df <- data.frame(rbind(colSums(enr_ann[,cazy_cols]), 
                            colSums(unenr_ann[,cazy_cols]), 
                            colSums(all_ann[,cazy_cols])))
rownames(cazy_df) <- c("enr", "unenr", "all")
colnames(cazy_df) <- gsub("CAZy  ", "", gsub("\\.", " ", colnames(cazy_df)))
wilcox.test(t(cazy_df[1,]), t(cazy_df[2,]), paired=TRUE) 
plot.new()
par(mfrow=c(1,1))
barplot(as.matrix(cazy_df[c(1,3),order(-cazy_df[3,])]), las=2, beside=T, xlab="Substrate targetted", ylab="Number of MAGs with function")
```


```{r}
# Figure 7
# Nitrogen cycle genes
betas <- merge(betas, ann, by="X")
nitrogen_ko <- c("K10944", "K10945", "K10946", "K10535","K00370", "K00371", "K00374", "K02567", "K02568", "K00368", "K15864", "K04561", "K02305", "K15877", "K00376", "K00362", "K00363", "K03385", "K15876", "K00367", "K10534", "K00372", "K00360", "K17877", "K00366", "K01915", "K00260", "K00264", "K00265", "K00266", "K15371", "K01183", "K03791", "K17525", "K00990", "K04751", "K07708", "K07712", "K03320", "K02586", "K02588", "K02591", "K00531", "K05916", "K01725", "K20932", "K20933", "K20934", "K20935", "K04752", "K07004", "K07273", "K01428", "K01429", "K01430")
nitrogen_gene <- c("amoA", "amoB", "amoC", "hao", "narG", "narH", "narI", "napA", "napB", "nirK", "nirS", "norB", "norC", "cyp55", "nosZ", "nirB", "nirD", "nrfA", "nrfH", "narB", "NR", "nasA", "nasB", "nit-6", "nirA", "glnA", "gudB", "glt1", "gltB", "gltD", "gdh2", "chit1", "putative_chitinase","CHID1", "glnD", "glnB", "glnL", "glnG", "amt", "nifD", "nifK", "nifH", "anfG", "hmp", "cynS", "hzsC", "hzsB", "hzsA", "hdh", "glnK","Xds", "lys", "ureA", "ureB", "ureC")
ko_to_gene_N <- read.csv("ko_to_gene_N.csv", header=T, stringsAsFactors = F)[,1:4]
betas_N <- betas[betas$kegg_id %in% ko_to_gene_N$KO,]
betas_grouped_N <- betas_N %>% group_by(kegg_id) %>% summarise(across(starts_with("betas"), ~ mean(.x, na.rm = TRUE)))
betas_grouped_N <- merge(betas_grouped_N, ko_to_gene_N, by.x="kegg_id", by.y="KO")
betas_cols <- which(colnames(betas_grouped_N) %like% "betas")
# Order by pathway
betas_grouped_N <- betas_grouped_N[order(betas_grouped_N$pathway),]

# N cycling genes by HMMs and peptidases identified by hotpep
enr_ann_hmm <- ann[ann$fasta %in% enr_ann$genome,]
enr_ann_hmm_N <- enr_ann_hmm %>% filter(kegg_id %in% nitrogen_ko & peptidase_id == "")
temp <- enr_ann_afe[,c(1, (ncol(enr_ann_afe)-12):ncol(enr_ann_afe))]
enr_ann_hmm_N <- merge(enr_ann_hmm_N, temp, by.y="genome", by.x="fasta")
enr_ann_hmm_pep <- enr_ann_hmm %>% filter(peptidase_id != "")
enr_ann_hmm_pep <- merge(enr_ann_hmm_pep, temp, by.y="genome", by.x="fasta")
enr_ann_hmm_N <- enr_ann_hmm_N[, -c(2:8, 11:23)]
enr_ann_hmm_N <- unique(enr_ann_hmm_N)
enr_ann_hmm_N$count24_100 <- 1
enr_ann_hmm_N$count48_100 <- 1
enr_ann_hmm_N$count72_100 <- 1
enr_ann_hmm_N$count168_100 <- 1
enr_ann_hmm_N$count48_50 <- 1
enr_ann_hmm_N$count168_50 <- 1
enr_ann_hmm_N$count24_100[enr_ann_hmm_N$AFE_24h==0] <- 0
enr_ann_hmm_N$count48_100[enr_ann_hmm_N$AFE_48h==0] <- 0
enr_ann_hmm_N$count72_100[enr_ann_hmm_N$AFE_72h==0] <- 0
enr_ann_hmm_N$count168_100[enr_ann_hmm_N$AFE_168h==0] <- 0
enr_ann_hmm_N$count48_50[enr_ann_hmm_N$AFE_48h_50==0] <- 0
enr_ann_hmm_N$count168_50[enr_ann_hmm_N$AFE_168h_50==0] <- 0

temp <- enr_ann_hmm_N[enr_ann_hmm_N$AFE_24h_100_col != "lightgrey",] %>% group_by(kegg_id) %>% tally(count24_100, name="h24")
N_df <- data.frame(matrix(nrow=nrow(temp), ncol=2))
N_df <- temp
temp <- enr_ann_hmm_N[enr_ann_hmm_N$AFE_72h_100_col != "lightgrey",] %>% group_by(kegg_id) %>% tally(count72_100, name="h72")
N_df <- merge(N_df, temp)
temp <- enr_ann_hmm_N[enr_ann_hmm_N$AFE_48h_100_col != "lightgrey",] %>% group_by(kegg_id) %>% tally(count48_100, name="h48")
N_df <- merge(N_df, temp)
temp <- enr_ann_hmm_N[enr_ann_hmm_N$AFE_168h_100_col != "lightgrey",] %>% group_by(kegg_id) %>% tally(count168_100, name="h168")
N_df <- merge(N_df, temp)
temp <- enr_ann_hmm_N[enr_ann_hmm_N$AFE_48h_50_col != "lightgrey",] %>% group_by(kegg_id) %>% tally(count48_50, name="h48_50")
N_df <- merge(N_df, temp)
temp <- enr_ann_hmm_N[enr_ann_hmm_N$AFE_168h_50_col != "lightgrey",] %>% group_by(kegg_id) %>% tally(count168_50, name="h168_50")
N_df <- merge(N_df, temp)
N_df <- N_df[order(-N_df$h24),]

par(mfrow=c(3,2))
barplot(N_df$h24, las=2, main="N pathways in MAGs enriched at 24h", ylim=c(0,80))
barplot(N_df$h48, las=2, main="N pathways in MAGs enriched at 72h", ylim=c(0,80))
barplot(N_df$h72, las=2, main="N pathways in MAGs enriched at 48h", ylim=c(0,80))
barplot(N_df$h168, las=2, main="N pathways in MAGs enriched at 168h", ylim=c(0,80))
barplot(N_df$h48_50, las=2, main="N pathways in MAGs enriched at 48h 50%", ylim=c(0,80))
barplot(N_df$h168_50, las=2, main="N pathways in MAGs enriched at 168h 50%", ylim=c(0,80))

N_df <- data.frame(N_df)
N_df_melt <- melt(N_df)
kruskal.test(value~variable, data=N_df_melt) # p=0.72

# Peptidases per genome per time point
enr_ann_hmm_pep$count24_100 <- 1
enr_ann_hmm_pep$count48_100 <- 1
enr_ann_hmm_pep$count72_100 <- 1
enr_ann_hmm_pep$count168_100 <- 1
enr_ann_hmm_pep$count48_50 <- 1
enr_ann_hmm_pep$count168_50 <- 1
enr_ann_hmm_pep$count24_100[enr_ann_hmm_N$AFE_24h==0] <- 0
enr_ann_hmm_pep$count48_100[enr_ann_hmm_N$AFE_48h==0] <- 0
enr_ann_hmm_pep$count72_100[enr_ann_hmm_N$AFE_72h==0] <- 0
enr_ann_hmm_pep$count168_100[enr_ann_hmm_N$AFE_168h==0] <- 0
enr_ann_hmm_pep$count48_50[enr_ann_hmm_N$AFE_48h_50==0] <- 0
enr_ann_hmm_pep$count168_50[enr_ann_hmm_N$AFE_168h_50==0] <- 0

pep_tally <- melt(enr_ann_hmm_pep[, c(1,35:40)])
pep_tally <- pep_tally %>% group_by(fasta, variable) %>% tally(value)
pep_tally <- data.frame(pep_tally)

# figure 7 D
ggboxplot(data=pep_tally, x="variable", y="n") + ylab("Peptidase genes per MAGs") + xlab("")
pep_tally$variable <- as.character(pep_tally$variable)
aovpep <- aov(n~variable, data=pep_tally)
summary(aovpep) # p=0.0005, F=4.5
tukeypep <- tukey_hsd(aovpep) # only significant p<0.05 differences are 168_50~48_100, 48_50~48_100, 72_100~48_50
# More peptidases per genome in the 50% samples

# N cycling gene counts and differential abundance in enriched, unenriched but detected and undetected MAGs
# The DRAM output has names of CAZy genes that I can pull from betas rows 435...
enr_ann <- read.delim("enr_MAGs_ann.tsv", sep="\t", header=T, stringsAsFactors = F)
#unenr_ann <- read.delim("unenr_MAGs_product.tsv", sep="\t", header=T, stringsAsFactors = F)
all_ann <- read.delim("annotations.tsv", sep="\t", header=T, stringsAsFactors = F)
pep_cols <- which(colnames(enr_ann) %like% "pept")
kegg_cols <- which(colnames(enr_ann) %like% "kegg")
enr_ann_N <- enr_ann[enr_ann$kegg_id %in% nitrogen_ko | enr_ann$cazy_hits %like% "chitinase",]
enr_ann_N <- merge(enr_ann_N, ko_to_gene_N, by.x="kegg_id", by.y="KO", all.x=T)
enr_ann_N$gene[enr_ann_N$kegg_id == ""] <- "Chitinase"
enr_ann_pep <- enr_ann[enr_ann$peptidase_id!="" & !(enr_ann$kegg_id %in% nitrogen_ko),]
all_ann_N <- all_ann[all_ann$kegg_id %in% nitrogen_ko | all_ann$cazy_hits %like% "chitinase",]
all_ann_N <- merge(all_ann_N, ko_to_gene_N, by.x="kegg_id", by.y="KO", all.x=T)
all_ann_N$gene[all_ann_N$kegg_id == ""] <- "Chitinase"
all_ann_pep <- all_ann[all_ann$peptidase_id!="" & !(all_ann$kegg_id %in% nitrogen_ko),]
inorN <- as.character(unique(all_ann_N$gene))
inorN <- inorN[which(!(inorN %in% c("lys", "ureA", "ureB", "ureC", "Xds", "chit1", "putative_chitinase")))]
N_df <- data.frame(matrix(ncol=length(inorN), nrow=2))
rownames(N_df) <- c("enr", "all")
colnames(N_df) <- inorN
# collapse multiple gene copies in the same genomes
inorN_enr <- enr_ann_N[enr_ann_N$gene %in% inorN, c(1,3,10,23:25)]
inorN_enr <- unique(inorN_enr)
inorN_all <- all_ann_N[all_ann_N$gene %in% inorN, c(1,3,10,23:25)]
inorN_all <- unique(inorN_all)
for (i in 1:ncol(N_df)) {
  N_df[1, i] <- nrow(inorN_enr[inorN_enr$gene == colnames(N_df[i]),])/1.14 #calculate % of enriched MAGs that have the gene at least once
  N_df[2, i] <- nrow(inorN_all[inorN_all$gene == colnames(N_df[i]),])/5.03 #calculate % of MAGs that have the gene at least once
}
t.test(t(N_df[1,]), t(N_df[2,]), paired=TRUE) 
plot.new()
par(mfrow=c(1,1))
barplot(as.matrix(N_df[,order(-N_df[2,])]), las=2, beside=T, xlab="Gene name", ylab="% of MAGs with function")
legend("topright", legend=c("Growing organisms", "Whole community"), fill=c("black", "lightgrey"))

orN <- c("lys", "ureA", "ureB", "ureC", "Xds", "chit1", "putative_chitinase", "Chitinase")
N_df <- data.frame(matrix(ncol=length(orN), nrow=2))
rownames(N_df) <- c("enr", "all")
colnames(N_df) <- orN
# collapse multiple gene copies in the same genomes
orN_enr <- enr_ann_N[enr_ann_N$gene %in% orN, c(1,3,10,23:25)]
orN_enr <- unique(orN_enr)
orN_all <- all_ann_N[all_ann_N$gene %in% orN, c(1,3,10,23:25)]
orN_all <- unique(orN_all)
for (i in 1:ncol(N_df)) {
  N_df[1, i] <- nrow(orN_enr[orN_enr$gene == colnames(N_df[i]),])/1.14 #calculate % of enriched MAGs that have the gene at least once
  N_df[2, i] <- nrow(orN_all[orN_all$gene == colnames(N_df[i]),])/5.03 #calculate % of MAGs that have the gene at least once
}
t.test(t(N_df[1,]), t(N_df[2,]), paired=TRUE) 
plot.new()
par(mfrow=c(1,1))
barplot(as.matrix(N_df[,order(-N_df[2,])]), las=2, beside=T, xlab="Gene name", ylab="% of MAGs with function")
legend("topright", legend=c("Growing organisms", "Whole community"), fill=c("black", "lightgrey"))

# Number of extracellular peptidase genes per genome - because almost all genomes have a peptidase
length(unique(enr_ann_pep$fasta)) #446/503 and 113/114
enr_ann_pep$peptidase_grp <- substring(enr_ann_pep$peptidase_family, 1, 1)
all_ann_pep$peptidase_grp <- substring(all_ann_pep$peptidase_family, 1, 1)
pep_per_mag_enr <- enr_ann_pep %>% group_by(fasta, peptidase_grp) %>% count()
pep_per_mag_all <- all_ann_pep %>% group_by(fasta, peptidase_grp) %>% count()
pep_per_mag_enr_grouped <- pep_per_mag_enr %>% group_by(peptidase_grp) %>% summarize_at("freq", sum)
pep_per_mag_all_grouped <- pep_per_mag_all %>% group_by(peptidase_grp) %>% summarize_at("freq", sum)
pep_df <- merge(pep_per_mag_enr_grouped, pep_per_mag_all_grouped, by="peptidase_grp", all=T)
pep_df[is.na(pep_df)] <- 0
pep_df <- pep_df[-which(pep_df$peptidase_grp %in% c("p", "I")),]
colnames(pep_df)[2:3] <- c("n_enr", "n_all")
pep_df$n_enr <- pep_df$n_enr/114
pep_df$n_all <- pep_df$n_all/503
t.test(pep_df$n_enr, pep_df$n_all, paired=TRUE) 
plot.new()
par(mfrow=c(1,1))
pep_df <- pep_df[order(-pep_df$n_all),]
rownames(pep_df) <- pep_df$peptidase_grp
barplot(as.matrix(t(pep_df[,-1])), las=2, beside=T, xlab="Peptidase Group", ylab="Mean number of peptidates per MAGs")
legend("topright", legend=c("Growing organisms", "Whole community"), fill=c("black", "lightgrey"))

```

```{r}
# Trehalose synthesis gene counts and differential abundance in enriched, unenriched but detected and undetected MAGs
trehalose <- c("K00697", "K16055", "K23377", "K02777", "K02819", "K01087", "K13057", "K01236", "K05343")
enr_ann <- read.delim("enr_MAGs_ann.tsv", sep="\t", header=T, stringsAsFactors = F)
all_ann <- read.delim("annotations.tsv", sep="\t", header=T, stringsAsFactors = F)
kegg_cols <- which(colnames(enr_ann) %like% "kegg")
enr_ann_T <- enr_ann[enr_ann$kegg_id %in% trehalose,]
all_ann_T <- all_ann[all_ann$kegg_id %in% trehalose,]
T_df <- data.frame(matrix(ncol=length(trehalose), nrow=2))
rownames(T_df) <- c("enr", "all")
colnames(T_df) <- trehalose
# collapse multiple gene copies in the same genomes
T_enr <- enr_ann_T[enr_ann_T$kegg_id %in% trehalose, c(2,9,10)]
T_enr <- unique(T_enr)
T_enr_50 <- T_enr[T_enr$fasta %in% afe50$bin,]
T_enr_100 <- T_enr[T_enr$fasta %in% afe100$bin,]
T_all <- all_ann_T[all_ann_T$kegg_id %in% trehalose, c(2,9,10)]
T_all <- unique(T_all)
for (i in 1:ncol(T_df)) {
  T_df[1, i] <- nrow(T_enr_50[T_enr_50$kegg_id == colnames(T_df[i]),])/(nrow(T_enr_50)/100) #calculate % of enriched MAGs that have the gene at least once
  T_df[2, i] <- nrow(T_enr_100[T_enr_100$kegg_id == colnames(T_df[i]),])/(nrow(T_enr_100)/100) #calculate % of MAGs that have the gene at least once
}
colnames(T_df)
t.test(t(T_df[1,]), t(T_df[2,]), paired=TRUE, alternative = "greater") 
wilcox.test(t(T_df[1,]), t(T_df[2,]), paired=TRUE, alternative = "greater")
plot.new()
par(mfrow=c(1,1))
barplot(as.matrix(T_df[,order(-T_df[2,])]), las=2, beside=T, xlab="Gene name", ylab="% of MAGs with function")
legend("topright", legend=c("Growing at 50%", "Growing at 100%"), fill=c("black", "lightgrey"))

# Diversity measures for 50% vs. 100%
enr_ann_50 <- enr_ann[enr_ann$fasta %in% afe50$bin,]
enr_ann_100 <- enr_ann[enr_ann$fasta %in% afe100$bin,]
length(unique(enr_ann_50$pfam))
length(unique(enr_ann_100$pfam))

# Detected at t0
mags_t0_50 <- rpkm[rpkm$X12C16O_0h_50_P09>0 | rpkm$X12C16O_0h_50_P10>0 | rpkm$X12C16O_0h_50_P13>0, "Genome"]
mags_t0_100 <- rpkm[rpkm$X12C16O_0h_100_P08>0 | rpkm$X12C16O_0h_100_P14>0 | rpkm$X12C16O_0h_100_P16>0, "Genome"]
all_ann_50_0 <- all_ann[all_ann$fasta %in% mags_t0_50,]
all_ann_100_0 <- all_ann[all_ann$fasta %in% mags_t0_100,]
length(unique(all_ann_50_0$pfam))
length(unique(all_ann_100_0$pfam))
pfam_0_50 <- all_ann_50_0 %>% count(pfam)
pfam_0_100 <- all_ann_100_0 %>% count(pfam)
H_0_50 <- diversity(pfam_0_50$n, "invsimpson")
H_0_100 <- diversity(pfam_0_100$n, "invsimpson")
# Detected at t48
mags_t48_50 <- rpkm[rpkm$X12C16O_48h_50_P09>0 | rpkm$X12C16O_48h_50_P10>0 | rpkm$X12C16O_48h_50_P13>0, "Genome"]
mags_t48_100 <- rpkm[rpkm$X12C16O_48h_100_P08>0 | rpkm$X12C16O_48h_100_P14>0 | rpkm$X12C16O_48h_100_P16>0, "Genome"]
all_ann_50_48 <- all_ann[all_ann$fasta %in% mags_t48_50,]
all_ann_100_48 <- all_ann[all_ann$fasta %in% mags_t48_100,]
length(unique(all_ann_50_48$pfam))
length(unique(all_ann_100_48$pfam))
pfam_48_50 <- all_ann_50_48 %>% count(pfam)
pfam_48_100 <- all_ann_100_48 %>% count(pfam)
H_48_50 <- diversity(pfam_48_50$n, "invsimpson")
H_48_100 <- diversity(pfam_48_100$n, "invsimpson")
# Detected at t168
mags_t168_50 <- rpkm[rpkm$X12C16O_168h_50_P09>0 | rpkm$X12C16O_168h_50_P10>0 | rpkm$X12C16O_168h_50_P13>0, "Genome"]
mags_t168_100 <- rpkm[rpkm$X12C16O_168h_100_P08>0 | rpkm$X12C16O_168h_100_P14>0 | rpkm$X12C16O_168h_100_P16>0, "Genome"]
all_ann_50_168 <- all_ann[all_ann$fasta %in% mags_t168_50,]
all_ann_100_168 <- all_ann[all_ann$fasta %in% mags_t168_100,]
length(unique(all_ann_50_168$pfam))
length(unique(all_ann_100_168$pfam))
pfam_168_50 <- all_ann_50_168 %>% count(pfam)
pfam_168_100 <- all_ann_100_168 %>% count(pfam)
H_168_50 <- diversity(pfam_168_50$n, "invsimpson")
H_168_100 <- diversity(pfam_168_100$n, "invsimpson")
```

```{r}
# Figure 8
# EPS synthesis gene counts and differential abundance in enriched, unenriched but detected and undetected MAGs
eps <- c("PF02563", "PF01451", "PF02706", "PF13807", "PF13614", "PF04932",  "PF01270", "PF06055", 
           "PF01943", "PF14667", "PF13440", "PF01554", "PF01061", "PF12698", "PF13414", "PF14524", 
           "PF05159", "PF02348", "PF01380", "PF00571", "PF13641", "PF07238", "PF13437", "PF08238", 
           "PF13372", "PF05048", "PF03062", "PF05426", "PF03170", "PF13432", "PF14559")
enr_ann$pfam <- gsub(".*\\[", "", gsub("].*", "", gsub("\\..*","", enr_ann$pfam_hits)))
all_ann$pfam <- gsub(".*\\[", "", gsub("].*", "", gsub("\\..*","", all_ann$pfam_hits)))
enr_ann_T <- enr_ann[enr_ann$pfam %in% eps,]
all_ann_T <- all_ann[all_ann$pfam %in% eps,]
T_df <- data.frame(matrix(ncol=length(eps), nrow=2))
rownames(T_df) <- c("enr", "all")
colnames(T_df) <- eps
# collapse multiple gene copies in the same genomes
T_enr <- enr_ann_T[enr_ann_T$pfam %in% eps, c(2,23)]
T_enr <- unique(T_enr)
T_all <- all_ann_T[all_ann_T$pfam %in% eps, c(2,23)]
T_all <- unique(T_all)
for (i in 1:ncol(T_df)) {
  T_df[1, i] <- nrow(T_enr[T_enr$pfam == colnames(T_df[i]),])/1.13 #calculate % of enriched MAGs that have the gene at least once
  T_df[2, i] <- nrow(T_all[T_all$pfam == colnames(T_df[i]),])/5.03 #calculate % of MAGs that have the gene at least once
}
t.test(t(T_df[1,]), t(T_df[2,]), paired=TRUE, alternative="greater") #p=0.002
wilcox.test(t(T_df[1,]), t(T_df[2,]), paired=TRUE, alternative="greater") #p=0.002
plot.new()
par(mfrow=c(1,1))
barplot(as.matrix(T_df[,order(-T_df[2,])]), las=2, beside=T, xlab="Gene name", ylab="% of MAGs with function")
legend("topright", legend=c("Growing organisms", "Whole community"), fill=c("black", "lightgrey"))
# Comparing precipitation treatments - not significant
T_enr_50 <- T_enr[T_enr$fasta %in% afe50$bin,]
T_enr_100 <- T_enr[T_enr$fasta %in% afe100$bin,]
for (i in 1:ncol(T_df)) {
  T_df[1, i] <- nrow(T_enr_50[T_enr_50$pfam == colnames(T_df[i]),])/(nrow(T_enr_50)/100) #calculate % of enriched MAGs that have the gene at least once
  T_df[2, i] <- nrow(T_enr_100[T_enr_100$pfam == colnames(T_df[i]),])/(nrow(T_enr_100)/100) #calculate % of MAGs that have the gene at least once
}
t.test(t(T_df[1,]), t(T_df[2,]), paired=TRUE) 
wilcox.test(t(T_df[1,]), t(T_df[2,]), paired=TRUE) 
plot.new()
par(mfrow=c(1,1))
barplot(as.matrix(T_df[,order(-T_df[2,])]), las=2, beside=T, xlab="Gene name", ylab="% of MAGs with function")
legend("topright", legend=c("50%", "100%"), fill=c("black", "lightgrey"))
# Compare time points
T_df <- data.frame(matrix(ncol=length(eps), nrow=6))
rownames(T_df) <- c("24h", "48h", "72h", "168h", "48h_50", "168h_50")
colnames(T_df) <- eps
# Get the gene content for MAGs that are actually growing at each time point
T_24_100 <- T_enr[T_enr$fasta %in% afe100$bin[afe100$AFE_24h>0],]
T_48_100 <- T_enr[T_enr$fasta %in% afe100$bin[afe100$AFE_48h>0],]
T_72_100 <- T_enr[T_enr$fasta %in% afe100$bin[afe100$AFE_72h>0],]
T_168_100 <- T_enr[T_enr$fasta %in% afe100$bin[afe100$AFE_168h>0],]
T_48_50 <- T_enr[T_enr$fasta %in% afe50$bin[afe100$AFE_48h>0],]
T_168_50 <- T_enr[T_enr$fasta %in% afe50$bin[afe100$AFE_168h>0],]
for (i in 1:ncol(T_df)) {
  T_df[1, i] <- nrow(T_24_100[T_24_100$pfam == colnames(T_df[i]),])/(length(unique(T_24_100$fasta))/100) #calculate % of enriched MAGs that have the gene at least once at 24h 100%
  T_df[2, i] <- nrow(T_48_100[T_48_100$pfam == colnames(T_df[i]),])/(length(unique(T_48_100$fasta))/100) #calculate % of enriched MAGs that have the gene at least once at 48h 100%
  T_df[3, i] <- nrow(T_72_100[T_72_100$pfam == colnames(T_df[i]),])/(length(unique(T_72_100$fasta))/100) #calculate % of enriched MAGs that have the gene at least once at 72h 100%
  T_df[4, i] <- nrow(T_168_100[T_168_100$pfam == colnames(T_df[i]),])/(length(unique(T_168_100$fasta))/100) #calculate % of enriched MAGs that have the gene at least once at 168h 100%
  T_df[5, i] <- nrow(T_48_50[T_48_50$pfam == colnames(T_df[i]),])/(length(unique(T_48_50$fasta))/100) #calculate % of enriched MAGs that have the gene at least once at 48h 50%
  T_df[6, i] <- nrow(T_168_50[T_168_50$pfam == colnames(T_df[i]),])/(length(unique(T_168_50$fasta))/100) #calculate % of enriched MAGs that have the gene at least once at 168h 50%
}
plot.new()
par(mfrow=c(1,1))
barplot(as.matrix(T_df[,order(-T_df[1,])]), las=2, beside=T, xlab="Gene name", ylab="% of MAGs with function", col=brewer.pal(6, "Paired"))
legend("topright", legend=c("Growing at 24h 100%", "Growing at 48h 100%", "Growing at 72h 100%", 
                            "Growing at 168h 100%", "Growing at 48h 50%", "Growing at 168h 50%"), 
       fill=brewer.pal(6, "Paired"))
eps_melt <- melt(t(T_df))
colnames(eps_melt) <- c("PFAM", "Time", "pct_MAGs")
eps_melt2 <- eps_melt[eps_melt$PFAM %in% c("PF13614", "PF01061", "PF13641", "PF00571", "PF01380"),]
ggboxplot(data=eps_melt2, x="Time", y="pct_MAGs", fill="Time")
eps_aov <- anova_test(pct_MAGs~Time, data=eps_melt)
summary(eps_aov) # Not significant
coefficients(eps_aov)
eps_aov <- aov(pct_MAGs~Time, data=eps_melt)
summary(eps_aov) # Significant, not sure why...is it because now it's doing paired comparisons?
# Maybe try in MAGs that are detected at 0h vs. whole community?
mags_t0 <- rpkm[rpkm$X12C16O_0h_100_P08>0 | rpkm$X12C16O_0h_100_P14>0 | rpkm$X12C16O_0h_100_P16>0, "Genome"]
t0_ann_T <- all_ann[all_ann$fasta %in% mags_t0 & all_ann$pfam %in% eps,]
all_ann_T <- all_ann[all_ann$pfam %in% eps,]
T_df <- data.frame(matrix(ncol=length(eps), nrow=2))
rownames(T_df) <- c("t0", "all")
colnames(T_df) <- eps
# collapse multiple gene copies in the same genomes
T_enr <- t0_ann_T[t0_ann_T$pfam %in% eps, c(2,23)]
T_enr <- unique(T_enr)
T_all <- all_ann_T[all_ann_T$pfam %in% eps, c(2,23)]
T_all <- unique(T_all)
for (i in 1:ncol(T_df)) {
  T_df[1, i] <- nrow(T_enr[T_enr$pfam == colnames(T_df[i]),])/(nrow(T_enr)/100) #calculate % of enriched MAGs that have the gene at least once
  T_df[2, i] <- nrow(T_all[T_all$pfam == colnames(T_df[i]),])/(nrow(T_all)/100) #calculate % of MAGs that have the gene at least once
}
t.test(t(T_df[1,]), t(T_df[2,]), paired=TRUE) 
plot.new()
par(mfrow=c(1,1))
barplot(as.matrix(T_df[,order(-T_df[2,])]), las=2, beside=T, xlab="Gene name", ylab="% of MAGs with function")
legend("topright", legend=c("Organisms detected at 0h", "Whole community"), fill=c("black", "lightgrey"))
# Following Kegg EPS biosynthesis pathway (two component system map)
eps <- c("K19667", "K01991", "K25307", "K16692", "K01791", "K02472", "K16712", "K16713")
eps_gene <- c("XpsR", "EpsA", "EpsP", "EpsB", "EpsC", "EpsD", "EpsE", "EpsF")
enr_ann_T <- enr_ann[enr_ann$kegg_id %in% eps,]
all_ann_T <- all_ann[all_ann$kegg_id %in% eps,]
T_df <- data.frame(matrix(ncol=length(eps), nrow=2))
rownames(T_df) <- c("enr", "all")
colnames(T_df) <- eps_gene
# collapse multiple gene copies in the same genomes
T_enr <- enr_ann_T[enr_ann_T$kegg_id %in% eps, c(2,9,10)]
T_enr <- unique(T_enr)
T_all <- all_ann_T[all_ann_T$kegg_id %in% eps, c(2,9,10)]
T_all <- unique(T_all)
for (i in 1:length(eps)) {
  T_df[1, i] <- nrow(T_enr[T_enr$kegg_id == eps[i],])/1.14 #calculate % of enriched MAGs that have the gene at least once
  T_df[2, i] <- nrow(T_all[T_all$kegg_id == eps[i],])/5.03 #calculate % of MAGs that have the gene at least once
}
t.test(t(T_df[1,]), t(T_df[2,]), paired=TRUE, alternative="greater")
plot.new()
par(mfrow=c(1,1))
barplot(as.matrix(T_df[,order(-T_df[2,])]), las=2, beside=T, xlab="Gene name", ylab="% of MAGs with function")
legend("topright", legend=c("Growing organisms", "Whole community"), fill=c("black", "lightgrey"))
T_enr_50 <- T_enr[T_enr$fasta %in% afe50$bin,]
T_enr_100 <- T_enr[T_enr$fasta %in% afe100$bin,]
for (i in 1:ncol(T_df)) {
  T_df[1, i] <- nrow(T_enr_50[T_enr_50$kegg_id == eps[i],])/(nrow(T_enr_50)/100) #calculate % of enriched MAGs that have the gene at least once
  T_df[2, i] <- nrow(T_enr_100[T_enr_100$kegg_id == eps[i],])/(nrow(T_enr_100)/100) #calculate % of MAGs that have the gene at least once
}
t.test(t(T_df[1,]), t(T_df[2,]), paired=TRUE) 
plot.new()
par(mfrow=c(1,1))
barplot(as.matrix(T_df[,order(-T_df[2,])]), las=2, beside=T, xlab="Gene name", ylab="% of MAGs with function")
legend("topright", legend=c("50%", "100%"), fill=c("black", "lightgrey"))
# Maybe try in MAGs that are detected at 0h vs. whole community?
mags_t0 <- rpkm[rpkm$X12C16O_0h_100_P08>0 | rpkm$X12C16O_0h_100_P14>0 | rpkm$X12C16O_0h_100_P16>0, "Genome"]
t0_ann_T <- all_ann[all_ann$fasta %in% mags_t0 & all_ann$kegg_id %in% eps,]
all_ann_T <- all_ann[all_ann$kegg_id %in% eps,]
T_df <- data.frame(matrix(ncol=length(eps), nrow=2))
rownames(T_df) <- c("t0", "all")
colnames(T_df) <- eps_gene
# collapse multiple gene copies in the same genomes
T_enr <- t0_ann_T[t0_ann_T$kegg_id %in% eps, c(2,9,10)]
T_enr <- unique(T_enr)
T_all <- all_ann_T[all_ann_T$kegg_id %in% eps, c(2,9,10)]
T_all <- unique(T_all)
for (i in 1:ncol(T_df)) {
  T_df[1, i] <- nrow(T_enr[T_enr$kegg_id == eps[i],])/(nrow(T_enr)/100) #calculate % of enriched MAGs that have the gene at least once
  T_df[2, i] <- nrow(T_all[T_all$kegg_id == eps[i],])/(nrow(T_all)/100) #calculate % of MAGs that have the gene at least once
}
t.test(t(T_df[1,]), t(T_df[2,]), paired=TRUE) 
plot.new()
par(mfrow=c(1,1))
barplot(as.matrix(T_df[,order(-T_df[2,])]), las=2, beside=T, xlab="Gene name", ylab="% of MAGs with function")
legend("topright", legend=c("Organisms detected at 0h", "Whole community"), fill=c("black", "lightgrey"))

```

```{r}
# Cellulose content analysis
cel_cont <- read.delim("Cellulose_content.txt", header=T, sep="\t")
colnames(cel_cont)
wilcox_test(data=cel_cont, Cellulose~precipitation, alternative = "two.sided") # p=0.4
```

